.github/workflows/publish_pypi.yml

```diff
@@ -8,13 +8,13 @@ on:
 
 jobs:
   build_and_upload:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@master
-    - name: Set up Python 3.7
+    - name: Set up Python 3.12
       uses: actions/setup-python@master
```

Comment:
I think this was casing the issue because in the master branch of setup-python they dropped Python 3.7.  Please change master to v5 so this would not happen again. v5 is currently the latest major version.